.github/ISSUE_TEMPLATE/simeng-bug-or-issue-report.md

```diff
@@ -0,0 +1,33 @@
+---
+name: SimEng Bug or Issue Report
+about: Report a bug or un-expected SimEng behaviour
+title: ''
+labels: ''
+assignees: ''
+
+---
+
+**Check List**
+ [ ] The binary I am trying to run has been compiled statically for either RV64 or AArch64.
+ [ ] The compiled binary is a Linux Elf file.
+ [ ] I have provided both a config file and a binary to SimEng as runtime arguments.
+
+**System Description**
+ - Operating System
+ - Compiler used to compile SimEng and version
+ - Compiler used to compile the static binary and version
+ - ISA binary is targetting (i.e. armv8.4-a+sve)
+ - Host Processor
+ - Main Memory Capacity
+
+**SimEng CMAKE Options Used**
+Provide a bullet list of all CMAKE options used. E.g. `-DCMAKE_BUILD_TYPE=Release`.
+
+**Binary Compilation Instructions**
+Provide a bullet list of how the binary in question was compiled, including all compiler flags used.
+
+**SimEng Command Line Expression**
+e.g. `./simeng /path/to/configs/a64fx.yaml /path/to/myBinary.elf`
+
```

Comment:
A lot of these questions could be answered by asking the submitter to provide the metadata from the run e.g.

```
/home/ra18837/PhD/repos/SimEng/cmake-build-debug/src/tools/simeng/simeng /home/ra18837/PhD/repos/SimEng/configs/DEMO_RISCV.yaml /home/ra18837/PhD/Programs/STREAM/stream.short.rv64gc
[SimEng] Build metadata:
[SimEng] 	Version: 0.9.6
[SimEng] 	Compile Time - Date: 14:01:44 - Jun 19 2024
[SimEng] 	Build type: Debug
[SimEng] 	Compile options: $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>;-Wall;-pedantic;-Werror
[SimEng] 	Test suite: ON

[SimEng] Running in Out-of-Order mode
[SimEng] Workload: /home/ra18837/PhD/Programs/STREAM/stream.short.rv64gc
[SimEng] Config file: /home/ra18837/PhD/repos/SimEng/configs/DEMO_RISCV.yaml
[SimEng] ISA: rv64
[SimEng] Auto-generated Special File directory: True
[SimEng] Special File directory used: /home/ra18837/PhD/repos/SimEng/cmake-build-debug/specialFiles/
[SimEng] Number of Cores: 1
[SimEng] Starting...

```

I guess there needs to be backup if they don't get that far though

Reply:
I think your final point here is important, and for those unfamiliar with SimEng the current format disects this up a bit easier

Reply:
You could ask for this/a full output as a way to confirm what they said. But yes, in addition to the current questions

Reply:
Added place for user to provide metadata printout